### ISSUE TITLE
Incorrect result for query that uses REGEXP_MATCHES

### ISSUE THREAD

**Issue by tianyangbest** [duckdb/duckdb#10239]
### What happens?

In duckdb query pandas, REGEXP_MATCHES function get incorrect result on chinese characters.

### To Reproduce

```sql
create table t as select '你好' as a;
select regexp_matches(a, '你好') as m from t;
```

except get true but get false.

only can reproduce in pandas and duckdb version 0.9.2 (v0.8.1 is correct) and **character must be chinese.**

### OS:

x86

### DuckDB Version:

0.9.2

### DuckDB Client:

python

### Full Name:

Yang Tian

### Affiliation:

University of Sydney

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi @tianyangbest, indeed this is a regression from 0.8.1. Thanks for reporting!

**Comment by szarnyasg** [comment]
@tianyangbest this is fixed on main (via PR #10061).
